contents/product-engineers/behavioral-analytics.mdx

```diff
@@ -0,0 +1,103 @@
+
+- Use a unique distinct ID.
+- Avoid using many different `distinct_id` values for the same user.
+- On the frontend, call identify as soon as possible (on load or login).
+- Reset on logout or when a user changes.
+
+As an engineer, it is up to you to get user identification right. Everyone relying on behavioral analytics (both technical and not) will assume the data is right, and from our experience, this goes wrong more than you think. Double-check and test your identification code. To be even more careful, tie it to your authorization code.
+
+> **Tip:** For examples of what "right" looks like, check out [our logic for identifying users](https://github.com/PostHog/posthog/blob/6b258dcaf67c03d31b72a3916a72fc893f0c883a/frontend/src/scenes/userLogic.ts#L124), look for other examples of `posthog.identify()` on [GitHub](https://github.com/search?q=posthog.identify+NOT+language%3APHP+NOT+language%3APython&type=code), or integrate with an auth provider like [PropelAuth](https://www.propelauth.com/post/posthog-propelauth).
+
+## 3. Utilize session replay
+
+One of the fastest time-to-insight behavioral analytics tools you can set up is [session replay](/session-replay). You might think of this as a tool for marketers and product managers, but engineers can benefit from it too.
+
+Replays focus on the behavior of an individual user. This gives them a high density of information. It isn't only a single event or an aggregate of events, rather all of their behavior visualized as they do it. This reveals gaps between expected experience and reality. 
+
+<ProductScreenshot
+  imageLight={ReplayLight} 
+  imageDark={ReplayDark} 
+  alt="Session replay in PostHog" 
+  classes="rounded"
+/>
+
+A unique benefit to engineers is that session replays are a great tool for debugging user issues. Instead of asking for information, you can see where the issue happened, its recreation, and get important context like person information, network performance, and console logs.
+
+> **Tip:** Add session replay links to your support tickets for easy access. We have a tutorial for how to do this with [Zendesk](/tutorials/zendesk-session-replays).
+
+## 4. Track releases and measure success
+
+If you're not careful, you can spend too much time looking at behavioral analytics (especially if it's heading in the right direction). Instead, you should plan to monitor it when it can make the most impact, and a prime time for that is feature launches.
+
+No matter how perfect you think your implementation is, there is always a chance it goes wrong. Users always seem to find ways to break or use features in unintended ways. Behavioral analytics enables you to monitor launches, visualize if something unexpected is happening, and pinpoint potential areas needing fixes. 
+
+PostHog is an especially good tool for this because it combines behavioral analytics tools like product analytics and session replay with developer tools like:
```

Comment:
```suggestion
PostHog is an especially good tool for this because it combines behavioral analytics tools, such as product analytics and session replay, with developer tools like:
```